include/gridtools/common/defs.hpp

```diff
@@ -27,13 +27,20 @@ namespace gridtools {
 #endif
 #endif
 
-#if defined(__has_builtin)
+#if defined(__NVCC__) && (__CUDACC_VER_MAJOR__ == 11 && __CUDACC_VER_MINOR__ >= 2 || __CUDACC_VER_MAJOR__ > 11)
+#define GT_ASSUME(x) __builtin_assume(x)
```

Comment:
```suggestion
#ifdef __CUDAARCH__
#define GT_ASSUME(x) __builtin_assume(x)
#endif
```
in case the host compiler doesn't support it (which is the case on daint ci), see the end of this section https://developer.nvidia.com/blog/boosting-productivity-and-performance-with-the-nvidia-cuda-11-2-c-compiler/#new_compiler_built-in_hints_for_better_device_code_optimizations
However this breaks your test.

Reply:
I believe that `__builtin_assume` only works for device code which gets generated when `__CUDA_ARCH__` is defined so instead of `__NVCC__` this needs to be `GT_CUDA_ARCH` or `__CUDA_ARCH__`.

Reply:
It's also ok outside of device code if the host compiler supports it, but that should be covered in the `__has_builtin` part. I think we should also keep the `__NVCC__`, because we specifically want that branch only for nvcc, not clang.